### PR TITLE
Feature/no committed status

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: application/models/testutils_test.go [hunk 1]
```diff
@@ -197,6 +235,38 @@ func createFileFixtures(n int) Files {
...
 }
 
+type potentialProvidersFixtures struct {
+	Users
+	Posts
+	PotentialProviders
+}
+
+// createPotentialProviderFixtures generates five PotentialProvider records for testing.
+// If necessary, four User and three Post fixtures will also be created.  The Posts will
```
Comment (Schparky) [inline application/models/testutils_test.go]:
```suggestion
// Four User and three Post fixtures will also be created.  The Posts will
```

File: application/templates/mail/request_from_committed_to_committed.plush.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<h4><a href="<%= postURL %>"><%= postTitle %></a></h4>
```
Comment (Schparky) [inline application/templates/mail/request_from_committed_to_committed.plush.html]: The name of this template doesn't match the content, but I assume that is what you're working on now.

File: application/actions/potentialprovider_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import (
+	"fmt"
+
+	"github.com/silinternational/wecarry-api/internal/test"
+)
+
+func (as *ActionSuite) Test_AddPotentialProvider() {
+
+	f := test.CreatePotentialProvidersFixtures(as.DB)
+	posts := f.Posts
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/actions/potentialprovider_test.go]: Good catch.

File: application/gqlgen/schema.graphql [hunk 4]
```diff
@@ -19,6 +19,9 @@ type Mutation {
     createPost(input: CreatePostInput!): Post!
     updatePost(input: UpdatePostInput!): Post!
     updatePostStatus(input: UpdatePostStatusInput!): Post!
+    addPotentialProvider(postID: String!): Post!
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/schema.graphql]: I'll go with `addMeAsPotentialProvider`.

File: application/gqlgen/post.go [hunk 5]
```diff
@@ -405,16 +421,83 @@ func (r *mutationResolver) UpdatePostStatus(ctx context.Context, input UpdatePos
...
+
+	return &post, nil
+}
+
+func (r *mutationResolver) RemoveMeAsPotentialProvider(ctx context.Context, postID string) (*models.Post, error) {
+	cUser := models.GetCurrentUserFromGqlContext(ctx)
+
+	var provider models.PotentialProvider
+
+	if err := provider.DestroyWithPostUUIDAndUserID(postID, cUser.ID, cUser); err != nil {
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/post.go]: Good idea.

File: application/templates/mail/request_from_committed_to_committed.plush.html (hunk 2 above)
Comment (Baggerone) [reply application/templates/mail/request_from_committed_to_committed.plush.html]: Yep.  Getting rid of that one and changing/adding others.

File: application/actions/potentialprovider_test.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+	qTemplate := `mutation {post: addPotentialProvider (postID: "%s")` +
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/actions/potentialprovider_test.go]: Hopefully I can remember that next time: :-/

File: application/models/potentialprovider.go [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+	var user User
+	if err := user.FindByID(userID); err != nil {
+		return errors.New("unable to find User in order to find PotentialProvider: " + err.Error())
+	}
+
+	if err := DB.Where("post_id = ? AND user_id = ?", post.ID, user.ID).First(p); err != nil {
+		return errors.New("unable to find PotentialProvider: " + err.Error())
+	}
+
+	return p.assertAuthorized(post, currentUser)
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/models/potentialprovider.go]: OK. I moved it over and it kind of required moving the contents of the Destroy method to the gql mutation, in order for it to have the post to send to the permission method.

File: application/actions/potentialprovider_test.go [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	as.Equal(want, resp.Post.PotentialProviders, "incorrect potential providers")
+}
+
+func (as *ActionSuite) Test_RemoveMeAsPotentialProvider() {
+
+	f := test.CreatePotentialProvidersFixtures(as.DB)
+	posts := f.Posts
+
+	qTemplate := `mutation {post: removeMeAsPotentialProvider (postID: "%s")` +
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/actions/potentialprovider_test.go]: :+1:

File: application/actions/potentialprovider_test.go [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+	want := []PotentialProvider{{ID: f.Users[3].UUID.String(), Nickname: f.Users[3].Nickname}}
+	as.Equal(want, resp.Post.PotentialProviders, "incorrect potential providers")
+}
+
+func (as *ActionSuite) Test_RemovePotentialProvider() {
+
+	f := test.CreatePotentialProvidersFixtures(as.DB)
+	posts := f.Posts
+
+	qTemplate := `mutation {post: removePotentialProvider (postID: "%s", userID: "%s")` +
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/actions/potentialprovider_test.go]: :+1:

File: application/models/post.go [hunk 10]
```diff
@@ -314,12 +305,40 @@ func (p *Post) NewWithUser(pType PostType, currentUser User) error {
 	return nil
 }
 
-func (p *Post) SetProviderWithStatus(status PostStatus, currentUser User) {
+func (p *Post) SetProviderWithStatus(status PostStatus, providerID *string) error {
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/models/post.go]: :+1:

File: application/models/potentialprovider.go [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+
+	return nil
+}
+
+// Update writes the PotentialProvider data to an existing database record.
+func (p *PotentialProvider) Update() error {
+	return update(p)
+}
+
+// FindByPostIDAndUserID reads a request record by the given Post ID and User ID
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/models/potentialprovider.go]: :+1:

File: application/notifications/dummy.go [hunk 12]
```diff
@@ -47,10 +47,6 @@ var dummyTemplates = map[string]dummyTemplate{
 		subject: domain.MessageTemplateRequestNotReceivedAfterAll,
 		body:    "The status of a request changed from completed to accepted or delivered.",
 	},
-	domain.MessageTemplateRequestFromAcceptedToCommitted: {
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/notifications/dummy.go]: :+1:

File: application/gqlgen/schema.graphql [hunk 13]
```diff
@@ -328,6 +331,7 @@ input LocationInput {
 input UpdatePostStatusInput {
     id: ID!
     status: PostStatus!
+    userID: ID
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/schema.graphql]: Good. Matt likes `providerUserID`

File: application/gqlgen/post.go [hunk 14]
```diff
@@ -405,16 +421,83 @@ func (r *mutationResolver) UpdatePostStatus(ctx context.Context, input UpdatePos
...
+	if err := post.SetProviderWithStatus(input.Status, input.UserID); err != nil {
+		return nil, reportError(ctx, errors.New("error setting provider with status: "+err.Error()),
+			"UpdatePostStatus.SetProvider", extras)
+	}
+
 	if err := post.Update(); err != nil {
 		return nil, reportError(ctx, err, "UpdatePostStatus", extras)
 	}
 
+	if err := post.DestroyPotentialProviders(input.Status, cUser); err != nil {
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/post.go]: I think I'll leave it here for now.  It should only be called if the Post's status has changed.

File: application/gqlgen/post.go [hunk 15]
```diff
@@ -405,16 +421,83 @@ func (r *mutationResolver) UpdatePostStatus(ctx context.Context, input UpdatePos
...
+	var provider models.PotentialProvider
+
+	if err := provider.DestroyWithPostUUIDAndUserUUID(postID, userID, cUser); err != nil {
+		return nil, reportError(ctx, errors.New("error removing potential provider: "+err.Error()),
+			"Post.RemovePotentialProvider")
+	}
+
+	var post models.Post
+	if err := post.FindByUUID(postID); err != nil {
+		return nil, reportError(ctx, err, "Post.RemovePotentialProvider")
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/post.go]: :+1:

File: application/gqlgen/post.go [hunk 16]
```diff
@@ -405,16 +421,83 @@ func (r *mutationResolver) UpdatePostStatus(ctx context.Context, input UpdatePos
...
+}
+
+func (r *mutationResolver) RemoveMeAsPotentialProvider(ctx context.Context, postID string) (*models.Post, error) {
+	cUser := models.GetCurrentUserFromGqlContext(ctx)
+
+	var provider models.PotentialProvider
+
+	if err := provider.DestroyWithPostUUIDAndUserID(postID, cUser.ID, cUser); err != nil {
+		return nil, reportError(ctx, errors.New("error removing potential provider: "+err.Error()),
+			"Post.RemoveMeAsPotentialProvider")
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/post.go]: :+1:

File: application/gqlgen/post.go [hunk 17]
```diff
@@ -405,16 +421,83 @@ func (r *mutationResolver) UpdatePostStatus(ctx context.Context, input UpdatePos
...
+	}
+
+	var post models.Post
+	if err := post.FindByUUID(postID); err != nil {
+		return nil, reportError(ctx, err, "Post.RemoveMeAsPotentialProvider")
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/post.go]: :+1:

File: application/gqlgen/post.go [hunk 18]
```diff
@@ -405,16 +421,83 @@ func (r *mutationResolver) UpdatePostStatus(ctx context.Context, input UpdatePos
...
+}
+
+func (r *mutationResolver) RemovePotentialProvider(ctx context.Context, postID, userID string) (*models.Post, error) {
+	cUser := models.GetCurrentUserFromGqlContext(ctx)
+
+	var provider models.PotentialProvider
+
+	if err := provider.DestroyWithPostUUIDAndUserUUID(postID, userID, cUser); err != nil {
+		return nil, reportError(ctx, errors.New("error removing potential provider: "+err.Error()),
+			"Post.RemovePotentialProvider")
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/gqlgen/post.go]: :+1:

**Pull request closed without merging**